services-geocoding/src/main/java/com/mapbox/api/geocoding/v6/models/V6Context.java

```diff
@@ -0,0 +1,113 @@
+package com.mapbox.api.geocoding.v6.models;
+
+import androidx.annotation.Nullable;
+
+import com.google.auto.value.AutoValue;
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * V6 response's piece of data which breaks out the complete geographical hierarchy
+ * for a given address or place. It is both a reliable way to access the named values of each
+ * component part of an address, plus contains feature-specific data such as the Wikidata id and
+ * 3-letter alpha country code.
+ *
+ * @see <a href="https://docs.mapbox.com/api/search/geocoding-v6/#the-context-object">The Context Object</a>
+ */
+@AutoValue
+public abstract class V6Context extends V6Object {
+
+  /**
+   * Address context element.
+   *
+   * @return address context element
+   */
+  @SerializedName("address")
+  @Nullable
+  public abstract V6ContextElement address();
+
+  /**
+   * Street context element.
+   *
+   * @return street context element
+   */
+  @SerializedName("street")
+  @Nullable
+  public abstract V6ContextElement street();
+
+  /**
+   * Neighborhood context element.
+   *
+   * @return neighborhood context element
+   */
+  @SerializedName("neighborhood")
+  @Nullable
+  public abstract V6ContextElement neighborhood();
+
+  /**
+   * Place context element.
+   *
+   * @return place context element
+   */
+  @SerializedName("place")
+  @Nullable
+  public abstract V6ContextElement place();
```

Comment:
Is this server-defined protocol? I mean that all of V6Context inner objects have the structure id + name? Because it's kinda counterintuitive that address and postcode have the same structure... Also, the "postcode name" sounds a bit weird.

Reply:
id and name are common parameters for all components. But you are right that the components will have different set of available additional properties. I didn't implemented this because it's still not in public api https://docs.mapbox.com/api/search/geocoding-v6/#the-context-object.

I think that in the next iteration `V6ContextElement` will be a base class and subclasses will have additional properties.

Reply:
Yeah, I can imagine what this name would mean for postcode, but still it's not exactly a name...
But if that's fixed in the json response, we should leave "name".

> I think that in the next iteration V6ContextElement will be a base class and subclasses will have additional properties.

Sounds good.